### ISSUE TITLE
SA1300 should check the names of local constants

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#1312]
Currently SA1300 does not check the names of variables which are marked `const`. These variables should be treated like other constants, and a violation reported if they do not start with an upper-case letter.

Example:

```csharp
public void Foo()
{
  const int x = 3; // SA1300
  Console.WriteLine(x);
}
```

**Comment by test20150828** [comment]
I don't like the idea to enforce local consts to be started with an upper-case letter. I never named local consts like this in my code. If an identifier starts with an upper-case letter then it always has an external scope (class field, method, etc.) and the upper-case-started local consts would not be consistent with this.

I would rather treat them like immutable variables and prefer to enforce them to be started with a lower-case letter instead (maybe it make sense to check them within the newly proposed rule #1310 and not to create a separate rule).

**Comment by sharwell** [comment]
@test20150828 Rule #1310 requires that local variables use the same naming convention as private fields. Like the original rule for private fields (SA1306), this rule does not evaluate variables marked `const`. This proposal is a consistent pairing with that rule - it applies the same naming convention for local `const` variables as is current applied for private `const` fields. Requiring that local `const` variables start with a lower case variable would introduce an inconsistency in our rules which I would like to avoid.

**Comment by sharwell** [comment]
I converted this to a new rule proposal.
